recipes/sfml/all/conanfile.py

```diff
@@ -0,0 +1,186 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class SFMLConan(ConanFile):
+    name = 'sfml'
+    description = 'Simple and Fast Multimedia Library'
+    license = "Zlib"
+    topics = ('conan', 'sfml', 'multimedia')
+    homepage = 'https://github.com/SFML/SFML'
+    url = 'https://github.com/conan-io/conan-center-index'
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+    generators = 'cmake', 'cmake_find_package'
+    settings = 'os', 'compiler', 'build_type', 'arch'
+    options = {
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'window': [True, False],
+        'graphics': [True, False],
+        'network': [True, False],
+        'audio': [True, False],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'window': True,
+        'graphics': True,
+        'network': True,
+        'audio': True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        if self.options.graphics:
+            self.requires('freetype/2.10.1')
+            self.requires('stb/20200203')
+        if self.options.audio:
+            self.requires('openal/1.20.1')
+            self.requires('flac/1.3.3')
+            self.requires('ogg/1.3.4')
+            self.requires('vorbis/1.3.6')
+        if self.options.window:
+            if self.settings.os in ['Linux', 'FreeBSD']:
+                self.requires('xorg/system')
+            self.requires('opengl/system')
+
+    def system_requirements(self):
+        if self.settings.os == 'Linux' and tools.os_info.is_linux:
+            if tools.os_info.with_apt:
+                installer = tools.SystemPackageTool()
+                packages = []
+                if self.options.window:
+                    packages.extend(['libudev-dev'])
+                for package in packages:
+                    installer.install(package)
```

Comment:
libudev desires a proxy package, libusb also requires libudev, but it's complete outdated.

Reply:
there is an old PR for libudev: https://github.com/conan-io/conan-center-index/pull/904

Reply:
```suggestion
    def system_requirements(self):
        if self.settings.os == "Linux" and tools.os_info.is_linux and self.options.window:
            package_tool = tools.SystemPackageTool(default_mode="verify")
            if tools.os_info.with_apt:
                packages = ["libudev-dev"]
            elif tools.os_info.with_yum:
                packages = ["libudev-devel"]
            elif tools.os_info.with_zypper:
                packages = ["libudev-devel"]
            elif tools.os_info.with_pacman:
                packages = ["libsystemd systemd"]
            else:
                packages = []
                self.output.warn("Don't know how to install libudev: Undefined package name for current platform.")
            for p in packages:
                package_tool.install(update=True, packages=p)
```